src/Material/Options.elm

```diff
@@ -135,24 +175,44 @@ collect1' options acc =
     Class x -> { acc | classes = x :: acc.classes }
     CSS x -> { acc | css = x :: acc.css }
     Attribute x -> { acc | attrs = x :: acc.attrs }
+    Internal x -> { acc | internal = x :: acc.internal }
     Many options -> List.foldl collect1' acc options
     Set _ -> acc 
+    Listener event options decoder ->
```

Comment:
Ordering might matter—try put higher-frequency cases first. Also, maybe tuck away Set _ and nop into a final `_ -> acc` case.